conformance/lang/expressions/literals/floating_point_literal.balt

```diff
@@ -0,0 +1,5665 @@
+    io:println(g); // @output 0
+
+    g = .00267848989D;
+    io:println(g); // @output 0.00267848989
+
+    g = .000267848989D;
+    io:println(g); // @output 0.000267848989
+
+    g = 7.406e+0;
+    io:println(g); // @output 7.406
+
+    g = 23676734.8E-0;
+    io:println(g); // @output 23676734.8
+
+    g = .3564E+0;
+    io:println(g); // @output 0.3564
+
+    g = .267848989e-0;
+    io:println(g); // @output 0.267848989
+
+    g = .0000E+12;
+    io:println(g); // @output 0
+
+    g = 7.406e+201;
+    io:println(g); // @output 7.406E+201
+
+    g = 23676734.8E-21;
+    io:println(g); // @output 2.36767348E-14
+
+    g = .3564E+12;
+    io:println(g); // @output 3.564E+11
+
+    g = .267848989e-124;
+    io:println(g); // @output 2.67848989E-125
+
+    g = 7.406e+0d;
+    io:println(g); // @output 7.406
+
+    g = 23676734.8E-0D;
+    io:println(g); // @output 23676734.8
+
+    g = .00001E+123d;
+    io:println(g); // @output 1E+118
+
+    g = .267848989e-14D;
+    io:println(g); // @output 2.67848989E-15
+
+    g = 0E+0;
+    io:println(g); // @output 0
+
+    g = 0e-0;
+    io:println(g); // @output 0
+
+    g = 7e+56;
+    io:println(g); // @output 7E+56
+
+    g = 7e-56;
+    io:println(g); // @output 7E-56
+
+    g = 785E+207;
+    io:println(g); // @output 7.85E+209
+
+    g = 785E-207;
+    io:println(g); // @output 7.85E-205
+
+    g = 7e+56d;
+    io:println(g); // @output 7E+56
+
+    g = 7e-56D;
+    io:println(g); // @output 7E-56
+
+    g = 785E+207D;
+    io:println(g); // @output 7.85E+209
+
+    g = 785E-207d;
+    io:println(g); // @output 7.85E-205
+
+    g = 25e1742d;
+    io:println(g); // @output 2.5E+1743
+
+    g = 25e1742D;
+    io:println(g); // @output 2.5E+1743
+
+    g = 25E1742d;
+    io:println(g); // @output 2.5E+1743
+
+    g = 25E1742D;
+    io:println(g); // @output 2.5E+1743
+
+    g = 25e174;
+    io:println(g); // @output 2.5E+175
+
+    g = 25E174;
+    io:println(g); // @output 2.5E+175
+
+    g = 25.0d;
+    io:println(g); // @output 25.0
+
+    g = 25.0D;
+    io:println(g); // @output 25.0
+
+    g = 25.0e174d;
+    io:println(g); // @output 2.50E+175
+
+    g = 25.0e174D;
+    io:println(g); // @output 2.50E+175
+
+    g = 25.0E174d;
+    io:println(g); // @output 2.50E+175
+
+    g = 25.0E174D;
+    io:println(g); // @output 2.50E+175
+
+    g = 25.0e+174d;
+    io:println(g); // @output 2.50E+175
+
+    g = 25.0e-174D;
+    io:println(g); // @output 2.50E-173
+
+    g = 25.0E+174d;
+    io:println(g); // @output 2.50E+175
+
+    g = 25.0E-174D;
+    io:println(g); // @output 2.50E-173
+
+    g = 25.0;
+    io:println(g); // @output 25.0
+
+    g = 25.0e174;
+    io:println(g); // @output 2.50E+175
+
+    g = 25.0E174;
+    io:println(g); // @output 2.50E+175
+
+    g = 25.0e+174;
+    io:println(g); // @output 2.50E+175
+
+    g = 25.0E-174;
+    io:println(g); // @output 2.50E-173
+}
+
+Test-Case: output
+Description: Test decimal floating point number with floating point type descriptor.
```

Comment:
Shall we specifically say union of float and decimal instead?

Reply:
[88661c0](https://github.com/ballerina-platform/ballerina-spec/pull/947/commits/88661c07a0aef05afed313e0e4f61f0e8f3fae6b)